flatten_data_to_anndata.py

```diff
@@ -0,0 +1,145 @@
+3. Updates the AnnData object with information from the JSON annotations and root keys.
+4. Writes the modified AnnData object to a specified output file.
+
+Usage:
+python script.py --json path/to/json_file.json --anndata path/to/anndata_file.h5ad --output path/to/output_file.h5ad
+"""
+import argparse
+import json
+from typing import Optional
+
+import anndata
+
+LABELSET_NAME = "name"
+
+LABELSET = "labelset"
+
+ANNOTATIONS = "annotations"
+
+CELL_IDS = "cell_ids"
+
+
+def read_json_file(file_path):
+    """
+    Reads and parses a JSON file into a Python dictionary.
+
+    Args:
+        file_path (str): The path to the JSON file.
+
+    Returns:
+        dict: The JSON data as a Python dictionary.
+
+    Returns None if the file does not exist or if there is an issue
+    parsing the JSON content.
+
+    Example:
+        json_data = read_json_file('path/to/your/file.json')
+        if json_data is not None:
+            # Use the parsed JSON data as a dictionary
+            print(json_data)
+    """
+    try:
+        with open(file_path, "r") as file:
+            data = json.load(file)
+            return data
+    except (FileNotFoundError, json.JSONDecodeError) as e:
+        print(f"Error reading JSON file: {e}")
+        return None
+
+
+def read_anndata_file(file_path: str) -> Optional[anndata.AnnData]:
+    """Load anndata object from a file.
+
+    Args:
+        file_path: The path to the file containing the anndata object.
+
+    Returns:
+        The loaded anndata object if successful, else None.
+    """
+    try:
+        anndata_obj = anndata.read_h5ad(file_path, backed="r")
+        return anndata_obj
+    except Exception as e:
+        print(f"An error occurred while loading the file: {e}")
+        return None
+
+
+def is_list_of_strings(var):
+    """
+    Check if a value is a list of strings.
+
+    Parameters:
+        var (list or any): The value to be checked.
+
+    Returns:
+        bool: True if the value is a list containing only string elements,
+              False otherwise.
+
+    """
+    return isinstance(var, list) and all(isinstance(item, str) for item in var)
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--json", required=True, help="Input JSON file path")
+    parser.add_argument("--anndata", required=True, help="Input AnnData file path")
+    parser.add_argument(
+        "--output",
+        help="Output AnnData file name (default: output.h5ad)",
+        default="output.h5ad",
+    )
+
+    args = parser.parse_args()
+    json_file_path = args.json
+    anndata_file_path = args.anndata
+    output_file_path = args.output
+
+    input_json = read_json_file(json_file_path)
+    input_anndata = read_anndata_file(anndata_file_path)
+
+    # obs
+    annotations = input_json[ANNOTATIONS]
+
+    for ann in annotations:
+        cell_ids = ann.get(CELL_IDS, [])
+
+        for k, v in ann.items():
+            if k == CELL_IDS:
```

Comment:
@ubyndr So, if k != CELL_IDS, you just ignore?

Shouldn't an error be thrown?

Reply:
Cell ids should be dropped while flattening to anndata, I'm just skipping them and continue doing flattening other keys

Reply:
The cell barcodes/cell ids must match both the JSON and the AnnData file. Otherwise, there's a warning and/or error thrown. 

@dosumis could explain

Reply:
The checks specified and implemented in the script here are pretty comprehensive for this https://github.com/cellannotation/castools/pull/4 

There is currently no modular architecture to pull those out.

Reply:
This PR is superseded by #8.  @evanbiederstedt 's criticism still applies. However, this should be fixed by running the checks that are currently in src/cas/anndata_conversion.py prior to flattening.